lib/stack_master/stack_dependency.rb

```diff
@@ -0,0 +1,65 @@
+module StackMaster
+  class StackDependency
+    StackOutputNotFound = Class.new(StandardError)
+
+    def initialize(stack_definition, config)
+      @stack_definition = stack_definition
+      @config = config
+    end
+
+    def outdated_stacks
+      @config.stacks.select do |stack|
+        dependent_stack = Stack.find(stack.region, stack.stack_name)
+        next unless dependent_stack
+        parameters = ParameterLoader.load(stack.parameter_files)
+        any_stack_output_outdated?(parameters, dependent_stack) || any_stack_outputs_outdated?(parameters, dependent_stack)
+      end
+    end
+
+    private
+
+    def any_stack_output_outdated?(params, stack)
+      params.any? do |_, value|
+        value['stack_output'] &&
+          value['stack_output'].gsub('_', '-') =~ %r(#{@stack_definition.stack_name}/) &&
+          outdated?(stack, value['stack_output'].split('/').last)
+      end
+    end
+
+    def any_stack_outputs_outdated?(params, stack)
+      params.any? do |key, value|
+        value['stack_outputs'] &&
+          value['stack_outputs'].any? do |output|
+            index = value['stack_outputs'].find_index(output)
+            dependent_parameter = stack_parameter(stack, key)
+            this_output_value = dependent_parameter.split(',')[index]
+            output.gsub('_', '-') =~ %r(#{@stack_definition.stack_name}/) &&
+              output_value(output.split('/').last.camelize) != this_output_value
```

Comment:
It's a slightly different case here because `value['stack_output'].split('/').last` does not return anything when we call `stack_parameter(dependent_stack, value['stack_output'].split('/').last)`.

When we use `stack_outputs` we define an input variables i.e. `subnets` and define multiple stack_outputs that form a comma separated list as the input. So:

```
Subnets:
  stack_outputs:
    - my_vpc/subnet_1
    - my_vpc/subnet_2
```

Results in an input parameter of:

```
Subnets: subnet-123abc,subnet456def
```

So we need to get that input parameter, split it and find one at the right index to compare to our output parameter.